app/controllers/api/connections_controller.rb

```diff
@@ -55,4 +64,104 @@ def request_match
 
     end
 
+    def update_match
+
+        if @current_user
+
+            request = MatchRequest.find_by(id: params[:id])
+
+            if request
+                #check if request exists
+
+                if request.status_is_pending?
+                    
+                    begin
+                        request.status = params[:status]
+                    rescue
+                        error = {"error": "Updated status is invalid"}
+                        render(json: error, status: 400)
+                        return
+                    end
+
+                    if request.status_is_pending?
+                        error = {"error": "Can't update request to pending"}
+                        render(json: error, status: 400)
+                        return
+
+                    elsif request.status_is_cancelled? and request.student_id != @current_user.id
+                        error = {"error": "Only the student can cancel the match request"}
+                        render(json: error, status: 400)
+                        return
+
+                    elsif (request.status_is_refused? or request.status_is_accepted?) and request.skill_class.teacher_id != @current_user.id
+                        error = {"error": "Only the teacher can answer the match request"}
+                        render(json: error, status: 400)
+                        return
+
+                    end
```

Comment:
If you have these many `if` blocks where you do the same thing (create a JSON payload with an `error` message, and render it with a 400), maybe create a method for that and just do `render_400(<error message>) and return`